S7.Net/PLC.cs

```diff
@@ -108,8 +108,14 @@ public bool IsConnected
         ///  If you use an external ethernet card, this must be set accordingly.</param>
         public Plc(CpuType cpu, string ip, Int16 rack, Int16 slot)
         {
-            IP = ip;
+            if (!Enum.IsDefined(typeof(CpuType), cpu))
+                throw new InvalidEnumArgumentException(nameof(cpu), (int) cpu, typeof(CpuType));
+
+            if (string.IsNullOrEmpty(ip))
+                throw new ArgumentException("IP address must valid.", nameof(ip));
+
             CPU = cpu;
+            IP = ip;
```

Comment:
Field IP is public _get_ only. It would make sense to make it IPAddress directly.
```csharp
public string IP { get; private set; }
```
to
```csharp
public IPAddress IP { get; private set; }
```
And in constructor to

```csharp
IP = IPAddress.Parse(ip);
```

That would caught invalid IP right at the beginning.

Reply:
I prefere not to break anything, if possible. Imho wrong IP address is not a major concern. If it's invalid, it will throw exception.

Reply:
Ok, makes sense.